pkg/controller/mutators/cache.go

```diff
@@ -6,22 +6,30 @@ import (
 	"github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
 )
 
+type mutatorStatus struct {
+	ingestion MutatorIngestionStatus
```

Comment:
Does MutatorIngestionStatus not have a way to indicate there is a conflict? Should it?

Reply:
TLDR:

Imagine a system with 10 total mutators, 4 mutators in conflict, and 2 mutators that failed to ingest due to problems interacting with API server.  Which of these sets of metric data is more useful?

_Option 1: Separate "in conflict" metric_

- ingestion status is `active`: 4
- ingestion status is `error`: 6
- mutators in conflict: 4

_Option 2: Incorporate conflict into ingestion status_

- ingestion status is `active`: 4
- ingestion status is `conflict`: 4
- ingestion status is `error`: 2

Actual Discussion:

A fair point.  The thing worth considering here is the difference between the existing `error` state and the third state we could create: `conflict`.

This _could_ potentially eclipse the new "in conflict" metric.  Is that a good thing?

Right now, a mutator that is "in conflict" would also be recorded as in an `error` state. 

Mutator's that fail to ingest for other reasons would be in an `error` state, but not in conflict.  Reasons that this could happen (besides conflicts) include:
- mutator is nil (doesn't seem like a likely case)
- errors when creating the byPodStatus
  - problems with interacting with the API server mostly.  Failing to connect, information is inconsistent with what's expected, etc.

With that info in mind, it seems that a conflict is the most likely problem a Mutator will encounter during ingestion.  We might then consider the `error` ingestion metric and the conflicts metric to be duplicative.  We could essentially hold the same information in a single metric (ingestion status) by allowing that metric to have a third state: `conflict`.

@maxsmythe, wdyt?  Do you see a downside to not representing a conflict as an ingestion `error` state, but rather as a new `conflict` state?

Reply:
I'm leaning toward conflict being a separate metric.

Since conflict is also an error, we probably also want it counted as an error state and there is no way to have multiple values for the same tag. A separate metric seems like a good compromise.

Reply:
SGTM! If we _do_ ever want something more complex (like multiple values per tag, or more ganular tags), we'd want a different implementation (like an error code or similar). But I think as-is is good for what we've currently got.